beacon-chain/simulator/service.go

```diff
@@ -23,7 +24,7 @@ var log = logrus.WithField("prefix", "simulator")
 type Simulator struct {
 	ctx                    context.Context
 	cancel                 context.CancelFunc
-	p2p                    types.P2P
+	p2p                    shared.P2P
```

Comment:
Why are we moving beacon node p2p to shared?

Reply:
that describes the interface for p2p interactions such as `Send`,`Subscribe`, we were only using it in the beacon node before but now with this pr since the validator clients also are using it , it made sense to move it to shared

Reply:
I see. The question is whether we should allow client to have these p2p interactions. That's an open question we should survey with the team.